test/internal/kubernetes/utils.go

```diff
@@ -247,6 +249,19 @@ func WaitForPodDeployment(ctx context.Context, clientset *kubernetes.Clientset,
 	return errors.Wrapf(retrier.Do(ctx, checkPodDeploymentFn), "could not wait for deployment %s", deploymentName)
 }
 
+func WaitForDeploymentToDelete(ctx context.Context, deploymentsClient typedappsv1.DeploymentInterface, d appsv1.Deployment) error {
+	assertDeploymentNotFound := func() error {
+		_, err := deploymentsClient.Get(ctx, d.Name, metav1.GetOptions{})
+		// only if the error is "isNotFound", do we say, the deployment is deleted
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return errors.New(fmt.Sprintf("expected isNotFound error when getting deployment, but got %+v", err))
+	}
+	retrier := retry.Retrier{Attempts: DeleteRetryAttempts, Delay: RetryDelay}
```

Comment:
nit: should we have/add/use a `DeleteRetryDelay = X `? Anything faster then 10 seconds would be nice. Can still do the math to have the total time polled equal to 60s

Reply:
Yes